pkg/agent/memberlist/cluster.go

```diff
@@ -200,18 +202,27 @@ func NewCluster(
 	return c, nil
 }
 
+func shouldJoinCluster(node *corev1.Node) bool {
+	// non-Linux Nodes should not join the memberlist cluster as all features relying on it is only supported on Linux.
+	return linuxNodeSelector.Matches(labels.Set(node.Labels))
+}
+
 func (c *Cluster) handleCreateNode(obj interface{}) {
 	node := obj.(*corev1.Node)
+	if !shouldJoinCluster(node) {
```

Comment:
It's possible but may be not worth as other modules need all Nodes so an unfiltered node informer is required anyway. If we create a filtered node informer, it will create another connection to kube-apiserver and store a copy of linux nodes:
```
// NewFilteredNodeInformer constructs a new informer for Node type.
// Always prefer using an informer factory to get a shared informer instead of getting an independent
// one. This reduces memory footprint and number of connections to the server.
```

Reply:
Makes sense